pkg/provider/passbolt/passbolt.go

```
@@ -0,0 +1,298 @@
+}
+
+type Client interface {
+	CheckSession(ctx context.Context) bool
+	Login(ctx context.Context) error
+	Logout(ctx context.Context) error
+	GetResource(ctx context.Context, resourceID string) (*api.Resource, error)
+	GetResources(ctx context.Context, opts *api.GetResourcesOptions) ([]api.Resource, error)
+	GetResourceType(ctx context.Context, typeID string) (*api.ResourceType, error)
+	DecryptMessage(message string) (string, error)
+	GetSecret(ctx context.Context, resourceID string) (*api.Secret, error)
+}
+
+func (provider *ProviderPassbolt) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube kclient.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	config := store.GetSpec().Provider.Passbolt
+
+	password, err := resolvers.SecretKeyRef(
+		ctx,
+		kube,
+		store.GetKind(),
+		namespace,
+		config.Auth.PasswordSecretRef,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	privateKey, err := resolvers.SecretKeyRef(
+		ctx,
+		kube,
+		store.GetKind(),
+		namespace,
+		config.Auth.PrivateKeySecretRef,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := api.NewClient(nil, "", config.Host, privateKey, password)
+	if err != nil {
+		return nil, err
+	}
+
+	provider.client = client
+	return provider, nil
+}
+
+func (provider *ProviderPassbolt) SecretExists(_ context.Context, _ esv1beta1.PushSecretRemoteRef) (bool, error) {
+	return true, nil
+}
+
+func (provider *ProviderPassbolt) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	if err := assureLoggedIn(ctx, provider.client); err != nil {
+		return nil, err
+	}
+
+	secret, err := getPassboltSecret(ctx, provider.client, ref.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	if ref.Property == "" {
+		return json.Marshal(secret)
+	}
+
+	return secret.GetProp(ref.Property)
+}
+
+func (provider *ProviderPassbolt) PushSecret(_ context.Context, _ *corev1.Secret, _ esv1beta1.PushSecretData) error {
+	return nil
```

Please return an error for unimplemented methods.